lib/datadog/core/remote/negotiation.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require_relative '../transport/http'
+
+module Datadog
+  module Core
+    module Remote
+      class Negotiation
```

Comment:
We should add unit tests for this class 😄

Reply:
Sure. I was hoping to have these tested by the `Component` spec but it made more sense to stub the `endpoint?` over there so now `Negotiation` is not tested.

Reply:
I will address that in a PR tomorrow

Reply:
PR: https://github.com/DataDog/dd-trace-rb/pull/2818